### ISSUE TITLE
src/sdk/vscode/lib/*.vsix does not exist

### ISSUE THREAD

**Issue by PhilippMundhenk** [GENIVI/iot-event-analytics#88]
According to the tldr, I should start by installing src/sdk/vscode/lib/*.vsix in Visual Studio code. This file does not exist.

**Comment by chheis** [comment]
Hi Philipp, it should be downloaded with executing "yarn".

**Comment by PhilippMundhenk** [comment]
I did execute yarn, but was not downloaded on my machine. Or maybe it was, but not to the location in src/... Which also makes sense, since it shouldn't be there...

**Comment by chheis** [comment]
did you check: .\src\sdk\vscode\lib\iotea-0.9.3.vsix  ?

**Comment by PhilippMundhenk** [comment]
I don't have a lib folder:
```bash
iot-event-analytics$ yarn
yarn install v1.22.10
[1/5] Validating package.json...
[2/5] Resolving packages...
warning Resolution field "underscore@1.12.1" is incompatible with requested version "underscore@~1.10.2"
success Already up-to-date.
Done in 0.84s.
iot-event-analytics$ ls src/sdk/vscode/
CHANGELOG.md  LICENSE  package.json  README.md  resources  src  test  tsconfig.dev.json  tsconfig.prod.json  yarn.lock
```

EDIT:
And `iot-event-analytics$ find . -name *.vsix` returns nothing

**Comment by llakie** [comment]
Run `yarn sdk.get` to download the SDKs. We removed that from the `postinstall` scripts, since this led to errors during Docker image creation. The documentation will be updated accordingly. (See #82 which is referring to of how it's build)

**Comment by PhilippMundhenk** [comment]
Does not seem to help much:
```bash
iot-event-analytics$ yarn sdk.get
yarn run v1.22.10
$ node get-sdks.js
Downloading library from https://github.com/GENIVI/iot-event-analytics/releases/download/vscode-ext-0.9.4/iotea-0.9.4.vsix...
Saving library to /home/mup2rng/sw/iot-event-analytics/src/sdk/vscode/lib/iotea-0.9.4.vsix...
Downloading library from https://github.com/GENIVI/iot-event-analytics/releases/download/py-sdk-0.4.1/boschio_iotea-0.4.1-py3-none-any.whl...
Saving library to /home/mup2rng/sw/iot-event-analytics/src/sdk/python/lib/boschio_iotea-0.4.1-py3-none-any.whl...
Downloading library from https://github.com/GENIVI/iot-event-analytics/releases/download/js-sdk-0.4.1/boschio.iotea-0.4.1.tgz...
Saving library to /home/mup2rng/sw/iot-event-analytics/src/sdk/javascript/lib/boschio.iotea-0.4.1.tgz...
ERROR: Received status 404 for GET https://github.com/GENIVI/iot-event-analytics/releases/download/js-sdk-0.4.1/boschio.iotea-0.4.1.tgz. Expected < 400
Done in 0.99s.
```

`yarn sdk.build` seems to work...

**Comment by PhilippMundhenk** [comment]
Build through VSCode Plugin still fails though, see also #93:
```bash
iot-event-analytics/docker-compose$ docker-compose -f "docker-compose.mosquitto.yml" -f "docker-compose.platform.yml" --project-name vscode-ext --env-file="iotea_helloworld/.env" up --build --remove-orphans
Building configmanager
Sending build context to Docker daemon  1.267MB
Step 1/26 : FROM node:12.13.0 as base
 ---> 11e92fc50c4a
Step 2/26 : FROM base as build
 ---> 11e92fc50c4a
Step 3/26 : ARG HTTP_PROXY
 ---> Using cache
 ---> e60514991243
Step 4/26 : ARG HTTPS_PROXY
 ---> Using cache
 ---> ab74251a3405
Step 5/26 : ENV HTTP_PROXY=${HTTP_PROXY}
 ---> Running in 95d5b0295659
Removing intermediate container 95d5b0295659
 ---> a7ffb7c9a74f
Step 6/26 : ENV HTTPS_PROXY=${HTTPS_PROXY}
 ---> Running in 42b134e8f2a4
Removing intermediate container 42b134e8f2a4
 ---> 19f2d2ad2d02
Step 7/26 : RUN mkdir /build
 ---> Running in 30ffd40f8ba9
Removing intermediate container 30ffd40f8ba9
 ---> 33a8599505ed
Step 8/26 : WORKDIR /build
 ---> Running in b855ca373cc9
Removing intermediate container b855ca373cc9
 ---> ba7a89343a1b
Step 9/26 : COPY package.json yarn.lock ./
 ---> a53373a4f546
Step 10/26 : RUN npm config set strict-ssl false -g
 ---> Running in cc154f248058
Removing intermediate container cc154f248058
 ---> 2788c6f0895b
Step 11/26 : RUN npm config set maxsockets 5 -g
 ---> Running in 085d0709fde4
Removing intermediate container 085d0709fde4
 ---> bb59c597fcad
Step 12/26 : RUN yarn --production
 ---> Running in d69665ee3542
yarn install v1.19.1
[1/5] Validating package.json...
[2/5] Resolving packages...
warning Resolution field "underscore@1.12.1" is incompatible with requested version "underscore@~1.10.2"
[3/5] Fetching packages...
[4/5] Linking dependencies...
[5/5] Building fresh packages...
Done in 44.61s.
Removing intermediate container d69665ee3542
 ---> 31a31f1f0620
Step 13/26 : RUN yarn audit
 ---> Running in 93be6916436f
yarn audit v1.19.1
warning Resolution field "underscore@1.12.1" is incompatible with requested version "underscore@~1.10.2"
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ moderate      │ Regular Expression Denial of Service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ ws                                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=5.2.3 <6.0.0 || >=6.2.2 <7.0.0 || >=7.4.6                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ ws                                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ ws                                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1748                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ moderate      │ Regular Expression Denial of Service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ ws                                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=5.2.3 <6.0.0 || >=6.2.2 <7.0.0 || >=7.4.6                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ mqtt                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ mqtt > ws                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1748                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ moderate      │ Regular expression denial of service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ glob-parent                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=5.1.2                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ mqtt                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ mqtt > help-me > glob-stream > glob-parent                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1751                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
3 vulnerabilities found - Packages audited: 420
Severity: 3 Moderate
Done in 1.66s.
The command '/bin/sh -c yarn audit' returned a non-zero code: 4
ERROR: Service 'configmanager' failed to build : Build failed
```

**Comment by llakie** [comment]
The 404 will be fixed within the next hour. The vulnerability issue will be resolved in an upcoming PR

**Comment by llakie** [comment]
We are now using the filter options of yarn audit to only fail on critical vulnerabilities. This should resolve these conflicts you are currently experiencing. Corresponding PR will be linked

**Comment by llakie** [comment]
Will be fixed in  PR #97

**Comment by llakie** [comment]
Merged
